.github/workflows/test-new-framework.yml

```
@@ -0,0 +1,47 @@
+      - name: Run new test framework
+        run: npm run test:new
+      - name: Check performance metrics
+        run: |
+          echo "New test framework performance:"
+          TIME=$(npm run test:new 2>&1 | grep "Duration" | sed 's/.*Duration \([0-9.]*\)ms.*/\1/')
+          echo "Execution time: ${TIME}ms"
+          if [ "${TIME}" -gt "30000" ]; then
+            echo "ERROR: Tests took longer than 30 seconds"
+            exit 1
+          fi
+          echo "✅ Performance target met"
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)